text2image.py

```diff
@@ -2,77 +2,51 @@
-)
-
-parser.add_argument(
-    "--output",
-    type=str,
-    nargs="?",
-    default="output.png",
-    help="where to save the output image",
-)
-
-parser.add_argument(
-    "--H",
-    type=int,
-    default=512,
-    help="image height, in pixels",
-)
-
-parser.add_argument(
-    "--W",
-    type=int,
-    default=512,
-    help="image width, in pixels",
-)
-
-parser.add_argument(
-    "--scale",
-    type=float,
-    default=7.5,
-    help="unconditional guidance scale: eps = eps(x, empty) + scale * (eps(x, cond) - eps(x, empty))",
-)
-
-parser.add_argument(
-    "--steps", type=int, default=50, help="number of ddim sampling steps"
-)
-
-parser.add_argument(
-    "--seed",
-    type=int,
-    help="optionally specify a seed integer for reproducible results",
-)
-
-parser.add_argument(
-    "--mp",
-    default=False,
-    action="store_true",
-    help="Enable mixed precision (fp16 computation)",
-)
+parser.add_argument("-p", "--prompt", type=str, nargs="?", default="post-apocalyptic wasteland, Thomas Kinkade, Colorful",
+    help="The prompt used to generate images")
+parser.add_argument("-o", "--output", type=str, nargs="?", default="output",
+    help="The prefix for the image file - will be appended with a number based on the number of copies")
+parser.add_argument("-H", "--height", type=int, default=512, help="Image height, in pixels")
+parser.add_argument("-W", "--width", type=int, default=512, help="Image width, in pixels")
+parser.add_argument("-s", "--scale", type=float, default=7.5,
+    help="Guidance scale. How closely the final output adheres to the prompt. The lower the value, the less adherence to the prompt. "
+         "Generally 0 to 15 or so but can go lower and higher. ")
+parser.add_argument("-t", "--steps", type=int, default=50, help="The number of DDIM sampling steps")
+parser.add_argument("-c", "--copies", type=int, default=1, help="The number of image copies to be created")
+parser.add_argument("-r", "--seed", type=int, help="The seed value to use for repeating a previous image generation - optional")
+parser.add_argument("--mp", default=False, action="store_true", help="Enable mixed precision (fp16 computation)")
 
 args = parser.parse_args()
-
 if args.mp:
     print("Using mixed precision.")
     keras.mixed_precision.set_global_policy("mixed_float16")
 
-generator = Text2Image(img_height=args.H, img_width=args.W, jit_compile=False)
-img = generator.generate(
+# Create output folder if it doens't exist
+if not isdir('output'):
+    os.mkdir('output')
+
+generator = Text2Image(img_height=args.height, img_width=args.width, jit_compile=False)
+
+seed, images = generator.generate(
     args.prompt,
     num_steps=args.steps,
     unconditional_guidance_scale=args.scale,
     temperature=1,
-    batch_size=1,
+    batch_size=args.copies,
     seed=args.seed,
 )
-Image.fromarray(img[0]).save(args.output)
-print(f"saved at {args.output}")
+print(f'The SEED for this batch is: {seed}')
+# Save results
+for img in images:
```

Comment:
```
for i, img in enumerate(images)
```

Reply:
The enumeration wouldn't achieve anything since the i value in the loop below is increased internally if an image with the same number is found 😄 Unless I'm missing something?

Reply:
oh i see. you'd want it to not clobber an existing image for some reason. got it.

Reply:
Yep, that's just for when you use the same output name but run multiple runs. Just a nice to have since I use the same prompt multiple times a day but look at the images at the end of the day 😄 Not necessarily something you'd want in the main codebase but wouldn't hurt?